.github/workflows/code-quality.yml

```diff
@@ -7,27 +7,52 @@ on:
       - "*.latest"
   pull_request:
   workflow_dispatch:
+    inputs:
+      dbt_adapters_branch:
+          description: "The branch of dbt-adapters to evaluate"
+          type: string
+          required: true
+          default: "main"
+  workflow_call:
+    inputs:
+      dbt_adapters_branch:
+        description: "The branch of dbt-adapters to evaluate"
+        type: string
+        required: true
+        default: "main"
 
 permissions: read-all
 
 # will cancel previous workflows triggered by the same event and for the same ref for PRs or same SHA otherwise
 concurrency:
   group: ${{ github.workflow }}-${{ github.event_name }}-${{ contains(github.event_name, 'pull_request') && github.event.pull_request.head.ref || github.sha }}
   cancel-in-progress: true
-
+env:
+  # set DBT_ADAPTERS_BRANCH to the input value if the event is a workflow_dispatch (workflow_call uses the same event_name),
+  # otherwise use 'main'
+  DBT_ADAPTERS_BRANCH: ${{ github.event_name == 'workflow_dispatch' && inputs.dbt_adapters_branch || 'main' }}
 jobs:
   code-quality:
     name: Code Quality
     runs-on: ubuntu-latest
 
+
     steps:
       - name: Check out repository
         uses: actions/checkout@v4
         with:
           persist-credentials: false
 
+      - name: Update Adapters and Core branches
+        if: ${{ github.event_name == 'workflow_call' || github.event_name == 'workflow_dispatch'}}
+        shell: bash
+        run: |
+          ./.github/scripts/update_dev_packages.sh \
+            ${{ github.event.inputs.dbt_adapters_branch }} \
+            "main"
+
       - name: Setup `hatch`
-        uses: dbt-labs/dbt-adapters/.github/actions/setup-hatch@main
+        uses: dbt-labs/dbt-adapters/.github/actions/setup-hatch@$DBT_ADAPTERS_BRANCH
```

Comment:
I think this becomes `${{ env.DBT_ADAPTERS_BRANCH }}` instead of `$DBT_ADAPTERS_BRANCH`. [From what it looks like](https://docs.github.com/en/actions/learn-github-actions/contexts#env-context), you can only omit the `env` syntax if it's in the `run` command.